pennylane/interfaces/batch/torch.py

```diff
@@ -93,11 +93,17 @@ def forward(ctx, kwargs, *parameters):  # pylint: disable=arguments-differ
                 break
 
         for i, r in enumerate(res):
-            if r.dtype is np.dtype("object"):
+            if isinstance(r, np.ndarray) and r.dtype is np.dtype("object"):
                 # For backwards compatibility, we flatten ragged tape outputs
                 r = np.hstack(r)
 
-            res[i] = torch.as_tensor(r, device=ctx.torch_device)
+            if isinstance(r, (list, tuple)):
+                res[i] = [torch.as_tensor(t) for t in r]
+
+                if isinstance(r, tuple):
+                    res[i] = tuple(res[i])
+            else:
+                res[i] = torch.as_tensor(r, device=ctx.torch_device)

```

Comment:
Fixes a bug where the Torch interface could not correctly cast to a Torch tensor the output of a QNode that has a mixture of sampling and expectation value outputs.